h/accounts/test/views_test.py

```diff
@@ -292,6 +293,264 @@ def test_logout_response_has_forget_headers(authn_policy):
     assert result.headers['x-erase-fingerprints'] == 'on the hob'
 
 
+forgot_password_fixtures = pytest.mark.usefixtures('activation_model',
+                                                   'authn_policy',
+                                                   'dummy_db_session',
+                                                   'form_validator',
+                                                   'mailer',
+                                                   'routes_mapper',
+                                                   'user_model')
+
+
+@forgot_password_fixtures
+def test_forgot_password_returns_error_when_validation_fails(authn_policy,
+                                                             form_validator):
+    request = DummyRequest(method='POST')
+    authn_policy.authenticated_userid.return_value = None
+    form_validator.return_value = ({"errors": "KABOOM!"}, None)
+
+    result = ForgotPasswordController(request).forgot_password()
+
+    assert result == {"errors": "KABOOM!"}
+
+
+@forgot_password_fixtures
+def test_forgot_password_fetches_user_by_form_email(authn_policy,
+                                                    form_validator,
+                                                    user_model):
+    request = DummyRequest(method='POST')
+    authn_policy.authenticated_userid.return_value = None
+    form_validator.return_value = (None, {"email": "giraffe@thezoo.org"})
+
+    ForgotPasswordController(request).forgot_password()
+
+    user_model.get_by_email.assert_called_with(request, "giraffe@thezoo.org")
```

Comment:
And I'd add that no single test can ensure that the code is right. That's why we don't write one test per function :). But if you made changes to `forgot_password` in such a way that `User.get_by_email` is never called, or is called with something other than the email address provided in the form, then I think that's a red flag and it's right that a test should start failing.